### PR TITLE
fix: fix toc2.js load failed with working in non-live notebook

### PR DESCRIPTION
**Pull request by zthxxx**
check all `IPython` and `Jupyter` call with `liveNotebook` flag

ref issuse: #1402

### CODE REVIEW

**Comment by zthxxx** [comment]
hi~ @jcb91 @juhasch  I make a pr again~ 

thanks for your awesome extensions! I try to fix a little bug in `toc2.js`

more reason details see #1403 

plz review and merge,  Thanks alot.

**Comment by zthxxx** [comment]
@jcb91 
Just a little change, 3 effective lines

**Comment by zthxxx** [comment]
@jcb91   Any suggestions about this pr?

**Comment by zthxxx** [comment]
@jcb91

File: src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js [hunk 1]
```diff
@@ -643,7 +648,7 @@
         });
 
         // update navigation menu
-        if (cfg.navigate_menu) {
+        if (cfg.navigate_menu && liveNotebook) {
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js]: I agree with that. That is the reason of the issue you encountered: if the navigate menu is enabled in the live notebook, then this config is stored in the notebook metadata and the js will try to create this menu even in the non live situation. 

Alternatively, we could also fix in hard the value `navigate_menu=false` in the toc2 template `toc2.tpl` which is used by nbconvert.

File: src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js [hunk 2]
```diff
@@ -56,6 +56,11 @@
      */
     var read_config = function () {
         var cfg = default_cfg;
+
+        if (!liveNotebook) {
+            return cfg;
+        }
+
```
Comment (jfbercher) [inline src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js]: This is not desirable. The idea (I am the author) was to be able to read the configuration stored in the notebook (line 66): the toc can be floating, the contents name may have been changed, etc

Comment (zthxxx) [reply src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js]: at line 66, the variable `IPython` referenced before assignment ([catch require do nothing with variable](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/7144c90048a33197755503064ab1a2a496480b11/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L84-L100))

Thus, the idea, read some config, is meaningless while in no-live-notebook

**Comment by zthxxx** [comment]
@jfbercher 

at [line 66](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/7144c90048a33197755503064ab1a2a496480b11/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L66), the variable IPython referenced before assignment (catch require do nothing with variable, [line 84-99](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/7144c90048a33197755503064ab1a2a496480b11/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L84-L99))

Thus, the idea, read some config, is meaningless while in no-live-notebook

**Comment by jfbercher** [comment]
Sorry, I was remembering falsely that the toc2 template (used for generating the html "non-live notebook" version) uses read_config, but that is not the case; It directly reads the config from notebook's metadata; So your analysis is correct and this fixes the mentioned issue. 

For completeness, it could also be useful to also add `cfg.navigate_menu=false;` in `toc2.tpl`  
```
        var cfg = {{ nb.get('metadata', {}).get('toc', {})|tojson|safe }};
        cfg.navigate_menu=false;
```
which also fixes the problem.

**Comment by jfbercher** [comment]
Thanks for this @zthxxx !

**Comment by zthxxx** [comment]
Thanks for your review! @jfbercher 

i think your solution is more fundamental.

**Pull request merged**